src/main/java/com/example/user_management/entity/User.java

```diff
@@ -3,19 +3,59 @@
 import jakarta.persistence.*;
 import lombok.Getter;
 import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.List;
 
 @Entity
 @Setter
 @Getter
-public class User {
+public class User implements UserDetails {
```

Comment:
Why entity implement UserDetails, this should be clean domain model representing mapping between DS and class and the class itself should not be aware of org.springframework.security package classes